app/containers/ChangeEmail/saga.js

```diff
@@ -0,0 +1,146 @@
+import { translationMessages } from 'i18n';
+import { takeLatest, put, call, select } from 'redux-saga/effects';
+
+import webIntegrationErrors from 'utils/web_integration/src/wallet/service-errors';
+
+import {
+  changeCredentialsInit,
+  changeCredentialsConfirm,
+  changeCredentialsComplete,
+  changeCredentialsGetKeysByPwd,
+} from 'utils/web_integration/src/wallet/change-credentials/change-credentials';
+
+import { makeSelectLocale } from 'containers/LanguageProvider/selectors';
+import { errorToastHandling } from 'containers/Toast/saga';
+
+import { logout } from 'containers/Logout/actions';
+
+import { selectEmail, selectVerificationCode } from './selectors';
+
+import {
+  CHANGE_EMAIL,
+  CHANGE_EMAIL_ERROR,
+  SEND_OLD_EMAIL,
+  SEND_OLD_EMAIL_ERROR,
+  CONFIRM_OLD_EMAIL,
+  CONFIRM_OLD_EMAIL_ERROR,
+  NEW_EMAIL_FIELD,
+  PASSWORD_FIELD,
+} from './constants';
+
+import {
+  changeEmailSuccess,
+  changeEmailErr,
+  sendOldEmailSuccess,
+  sendOldEmailErr,
+  confirmOldEmailSuccess,
+  confirmOldEmailErr,
+} from './actions';
+
+export function* sendOldEmailWorker({ resetForm, email }) {
+  try {
+    const locale = yield select(makeSelectLocale());
+    const translations = translationMessages[locale];
+
+    const response = yield call(() => changeCredentialsInit(email));
+
+    if (!response.OK) {
+      throw new Error(
+        translations[webIntegrationErrors[response.errorCode].id],
```

Comment:
What will happen here is webIntegrationErrors does not contain a value for error code?
do you think it makes sense to create a method that returns a translation for error code; and returns some default message if there is nothing for that error code?